gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/task/COPCMessageHandler.java

```diff
@@ -182,9 +195,18 @@ private void addLogForEarlyFragmentAndStore(InboundMessage inboundMessage, Strin
     }
 
     private void storeCOPCAttachment(PatientAttachmentLog fragmentAttachmentLog, InboundMessage inboundMessage,
-        String conversationId) throws ValidationException, InlineAttachmentProcessingException {
-        attachmentHandlerService.storeAttachementWithoutProcessing(fragmentAttachmentLog.getFilename(),
-            inboundMessage.getAttachments().get(0).getPayload(), conversationId, fragmentAttachmentLog.getContentType());
+                                     String conversationId) throws ValidationException, InlineAttachmentProcessingException {
+
+        if (fragmentAttachmentLog.getLargeAttachment() == null || fragmentAttachmentLog.getLargeAttachment()) {
+            attachmentHandlerService.storeAttachementWithoutProcessing(fragmentAttachmentLog.getFilename(),
+                    inboundMessage.getAttachments().get(0).getPayload(), conversationId, fragmentAttachmentLog.getContentType());
+        } else {
+            var attachment = attachmentHandlerService.buildInboundAttachmentsFromAttachmentLogs(
+                    Arrays.asList(fragmentAttachmentLog),
+                    Arrays.asList(inboundMessage.getAttachments().get(0).getPayload())
```

Comment:
IntelliJ suggests `List.of()` could be more suitable instead of `Arrays.asList()`